tuf/store/httpstore.go

```diff
@@ -270,13 +275,54 @@ func (s HTTPStore) buildURL(uri string) (*url.URL, error) {
 	return s.baseURL.ResolveReference(sub), nil
 }
 
-// GetKey retrieves a public key from the remote server
-func (s HTTPStore) GetKey(role string) ([]byte, error) {
+// GetKey retrieves the most recently created (whether it is signed in yet or not)
+// public key for the given role from the remote server
+func (s HTTPStore) GetKey(role string) (data.PublicKey, error) {
+	return s.requestKey(role, nil)
+}
+
+// RotateKey rotates a key on the remote server and returns the new public key.  This requires
+// a request with a short expiry time (to limit replay), signed by at least one root key.
+// Signing with the root key proves that the client making the request has the capability of
+// actually rotating the key, and is not just making a spurious rotation request.
+func (s HTTPStore) RotateKey(role string, cs signed.CryptoService, roots ...data.PublicKey) (data.PublicKey, error) {
+	req := &data.SignedCommon{
+		Type:    role,
+		Expires: time.Now().Add(5 * time.Minute),
+	}
+	reqJSON, err := json.Marshal(req)
+	if err != nil {
+		return nil, err
+	}
+	signedReq := &data.Signed{Signed: reqJSON}
+	if err := signed.Sign(cs, signedReq, roots...); err != nil {
```

Comment:
Yeah, the other store methods take or return a complete body (ok, for SetMulti it takes a map of part name to body). This method has to actually construct a body from the parameters it's passed.

If the signing was in an HTTP header or something more generic that could be included on any request, I'd feel better about putting it into the store.